backend/app/services/weather_retriever.rb

```diff
@@ -14,47 +14,82 @@ def get(date, postal_code)
       end
 
       forecast = get_forecast(date, position)
-      the_day = forecast&.daily&.data&.first
 
-      if the_day.blank?
-        Rails.logger.warn "No forecast found for position #{position.inspect}: #{forecast.inspect}"
+      if forecast.status != 200
+        Rails.logger.warn "No forecast found for position #{position.inspect}: response code was #{forecast.status}, headers were #{forecast.headers}, body contained #{forecast.body}"
 
         return
       end
 
-      create_weather(the_day, position.id)
+      create_weather(forecast, position.id)
     end
 
     private
 
     def get_forecast(date, position)
-      tz = Time.zone
-      Time.zone = NearestTimeZone.to(position.latitude, position.longitude)
-
-      forecast = ForecastIO.forecast(
-        position.latitude,
-        position.longitude,
-        time: Time.zone.parse(date.to_s).to_i,
-        params: {exclude: "currently,minutely,hourly,alerts,flags"}
+      forecast = Tomorrowiorb.forecast(
+        "#{position.latitude},#{position.longitude}",
+        ["1d"],
+        "metric"
       )
 
-      Time.zone = tz
-
       forecast
     end
 
-    def create_weather(the_day, position_id)
+    def create_weather(forecast, position_id)
+      today = forecast.body["timelines"]["daily"][0]
```

Comment:
nit: maybe use "dig" for json safety?

Reply:
Thanks, I'm pretty unfamiliar with Ruby and did not know about [dig](https://docs.ruby-lang.org/en/3.0/dig_methods_rdoc.html), I will look into that further and make some changes.